fdbserver/LeaderElection.actor.cpp

```diff
@@ -27,15 +27,33 @@
 
 Optional<std::pair<LeaderInfo, bool>> getLeader( const vector<Optional<LeaderInfo>>& nominees );
 
-ACTOR Future<Void> submitCandidacy( Key key, LeaderElectionRegInterface coord, LeaderInfo myInfo, UID prevChangeID, Reference<AsyncVar<vector<Optional<LeaderInfo>>>> nominees, int index ) {
+ACTOR Future<Void> submitCandidacy(Key key, LeaderElectionRegInterface coord, LeaderInfo myInfo, UID prevChangeID,
+                                   AsyncTrigger* nomineeChanged, Optional<LeaderInfo>* leaderInfo,
+                                   bool reportCoordConnFailure) {
+	state Optional<LeaderInfo> li;
 	loop {
-		auto const& nom = nominees->get()[index];
-		Optional<LeaderInfo> li = wait( retryBrokenPromise( coord.candidacy, CandidacyRequest( key, myInfo, nom.present() ? nom.get().changeID : UID(), prevChangeID ), TaskPriority::CoordinationReply ) );
+		if (reportCoordConnFailure) {
+			ErrorOr<Optional<LeaderInfo>> rep = wait(coord.candidacy.tryGetReply(
+			    CandidacyRequest(key, myInfo, leaderInfo->present() ? leaderInfo->get().changeID : UID(), prevChangeID),
+			    TaskPriority::CoordinationReply));
+			if (rep.isError() && rep.getError().code() == error_code_connection_failed) {
+				// connecting to this coordinator failed, most likely due to timeout.
+				throw coordinators_changed();
```

Comment:
Need to reset the connection file before throw this error. Otherwise the re-resolve will not happen.